crates/pathfinder/src/rpc/types.rs

```diff
@@ -83,29 +79,26 @@ pub mod relaxed {
 
     impl From<crate::sequencer::reply::Call> for Vec<H256> {
         fn from(call: crate::sequencer::reply::Call) -> Self {
-            call.result.into_iter().map(H256::from).collect()
+            call.result
+                .into_iter()
+                .map(|r| types::H256::from(r.0.to_be_bytes()).into())
```

Comment:
```suggestion
                .map(|r| H256(r.0.to_be_bytes()))
```

Reply:
I'm not actually sure we need `types::H256` at all anymore?

Reply:
That said, why are we casting to `H256` at all for these? Where is it used? I'd expect `H256` and `U256` to be restricted to the `ethereum` sections.

Reply:
Hmm call result has its stark hash based newtype, lemme check what is going on here.